plugins/askrene/mcf.c

```diff
@@ -1375,7 +1375,27 @@ linear_routes(const tal_t *ctx, struct route_query *rq,
 	struct reserve_hop *reservations = new_reservations(working_ctx, rq);
 
 	while (!amount_msat_is_zero(amount_to_deliver)) {
-                new_flows = tal_free(new_flows);
+		u32 num_parts, parts_slots, excess_parts;
+
+                /* FIXME: This algorithm to limit the number of parts is dumb
+                 * for two reasons:
+                 *      1. it does not take into account that several loop
+                 *      iterations here may produce two flows along the same
+                 *      path that after "squash_flows" become a single flow.
+                 *      2. limiting the number of "slots" to 1 makes us fail to
+                 *      see some solutions that use more than one of those
+                 *      existing paths.
+                 *
+                 * A better approach could be to run MCF, remove the excess
+                 * paths and then recompute a MCF on a network where each arc is
+                 * one of the previously remaining paths, ie. redistributing the
+                 * payment amount among the selected paths in a cost-efficient
+                 * way. */
+		new_flows = tal_free(new_flows);
+		num_parts = tal_count(*flows);
+		assert(num_parts < rq->maxparts);
+		parts_slots = rq->maxparts - num_parts;
+		excess_parts = 0;
```

Comment:
Don't initialize excess_parts gratuitously; I'd prefer the compiler to warn (as it will on higher optimization levels) if a path below doesn't initialize it.

Reply:
Got it.